### PR TITLE
Tutorial kool typed dynamic

### PR DESCRIPTION
**Pull request by daejunpark**

### CODE REVIEW

File: k-distribution/tutorial/2_languages/2_kool/2_typed/1_dynamic/kool-typed-dynamic.k [hunk 1]
```diff
@@ -46,7 +46,11 @@ untyped KOOL.  */
                 | Type "[" "]"
                 > Types "->" Type
                 | "(" Type ")"           [bracket]
+  // TODO(KORE): drop klabel once issues #1913 are fixed
+  syntax Types ::= List{Type,","}   [klabel(_,_::Types)]
```
Comment (grosu) [inline k-distribution/tutorial/2_languages/2_kool/2_typed/1_dynamic/kool-typed-dynamic.k:50]: this has been fixed today, hasn't it?

Comment (daejunpark) [inline k-distribution/tutorial/2_languages/2_kool/2_typed/1_dynamic/kool-typed-dynamic.k:50]: no, it is not for this case.

File: k-distribution/tutorial/2_languages/2_kool/2_typed/1_dynamic/kool-typed-dynamic.k [hunk 2]
```diff
@@ -404,8 +413,11 @@ Like in untyped KOOL. */
 The only change to untyped KOOL's values is that method closures are
 now typed (their first argument holds their type): */
 
+
  syntax Val ::= objectClosure(CrntObjCellFragment)
               | methodClosure(Type,Id,Int,Params,Stmt)
+ // TODO(KORE): drop the additional production once parsing issue #1842 is fixed
+              | objectClosure(K)
```
Comment (grosu) [inline k-distribution/tutorial/2_languages/2_kool/2_typed/1_dynamic/kool-typed-dynamic.k:415]: are you sure you still need this?  it worked for me to drop them in SIMPLE, two of them

File: k-distribution/tutorial/2_languages/2_kool/2_typed/1_dynamic/kool-typed-dynamic.k [hunk 3]
```diff
@@ -427,12 +439,14 @@ method returns.  We have to do the same with the current object
 environment, so we group them together in the stack frame.  */
 
   syntax KItem ::= (Map,K,ControlCellFragment)
+ // TODO(KORE): drop the additional production once parsing issue #1842 is fixed
+                 | (Map,K,K)
```
Comment (grosu) [inline k-distribution/tutorial/2_languages/2_kool/2_typed/1_dynamic/kool-typed-dynamic.k:438]: this one definitely was dropped in SIMPLE without any problem

File: k-distribution/tutorial/2_languages/2_kool/2_typed/1_dynamic/kool-typed-dynamic.k [hunk 4]
```diff
@@ -532,11 +546,15 @@ Like in untyped KOOL. */
 Like in untyped KOOL. */
 
   rule <k> class Class1 extends Class2 { S } => . ...</k>
+       // TODO(KORE): classes hack to avoid not having AC -- see 2230bb3947c34b34b09e44eaa2ea2dfbd46b31cf
```
Comment (grosu) [inline k-distribution/tutorial/2_languages/2_kool/2_typed/1_dynamic/kool-typed-dynamic.k]: is this still needed?  we do have AC now.

File: k-distribution/tutorial/2_languages/2_kool/2_typed/1_dynamic/kool-typed-dynamic.k (hunk 3 above)
Comment (daejunpark) [inline k-distribution/tutorial/2_languages/2_kool/2_typed/1_dynamic/kool-typed-dynamic.k:438]: but it cannot be dropped for KOOL

File: k-distribution/tutorial/2_languages/2_kool/2_typed/1_dynamic/kool-typed-dynamic.k [hunk 5]
```diff
@@ -532,11 +546,15 @@ Like in untyped KOOL. */
...
+       <classes>... (.Map => Class1 |-> ClassBody(Class2, S))
+       /*
        <classes>... (. => <class>
```
Comment (grosu) [inline k-distribution/tutorial/2_languages/2_kool/2_typed/1_dynamic/kool-typed-dynamic.k]: doesn't this `.` need to be a `.Bag` ?

File: k-distribution/tutorial/2_languages/2_kool/2_typed/1_dynamic/kool-typed-dynamic.k [hunk 6]
```diff
@@ -628,34 +652,34 @@ Like in untyped KOOL. */
   rule <k> X:Id => this . X ...</k> <env> Env:Map </env>
     when notBool(X in keys(Env))  [structural]
 
-  context HOLE . _ when (HOLE =/=K super)
+  context HOLE . _::Id when (HOLE =/=K super)
```
Comment (grosu) [inline k-distribution/tutorial/2_languages/2_kool/2_typed/1_dynamic/kool-typed-dynamic.k]: when => requires

File: k-distribution/tutorial/2_languages/2_kool/2_typed/1_dynamic/kool-typed-dynamic.k (hunk 1 above)
Comment (grosu) [inline k-distribution/tutorial/2_languages/2_kool/2_typed/1_dynamic/kool-typed-dynamic.k:50]: #2099?  It solved all the similar cases in SIMPLE for me.  Are you saying we need to revisit it?

Comment (daejunpark) [inline k-distribution/tutorial/2_languages/2_kool/2_typed/1_dynamic/kool-typed-dynamic.k:50]: Somehow, #2099 doesn't work for this. Yes, I'm saying that we need to revisit it. @cos Can you take a look?

**Comment by daejunpark** [comment]
@grosu please review. Two issues are still there, though.

**Comment by grosu** [comment]
I am merging, but hopefully we can fix those issues soon, too, at lest the first one.

**Pull request merged**